server/raft.go

```diff
@@ -1788,7 +1784,7 @@ func (n *raft) processAppendEntries() {
 }
 
 func (n *raft) runAsFollower() {
-	for {
+	for n.State() == Follower {
```

Comment:
This just pedantic or did you see a case where we could be running here not as a follower?

Reply:
In particular it seems we can get stuck in `runAsLeader`. I added the check to all three states just to be sure.

Reply:
Meaning we are in runAsLeader but not the leader? Do we have a test to reproduce?

Reply:
I am not sure if indefinitely or whether it's just taking a while for `runAsLeader` to stop and switch to another, but changing `for n.State() == Leader {` back to `for {` causes the new test in the PR to fail.

Reply:
ok thanks for the explanation.